filebeat/registrar/registrar.go

```diff
@@ -188,7 +190,9 @@ func (r *Registrar) Run() {
 
 		beforeCount := r.states.Count()
 		r.states.Cleanup()
```

Comment:
Consider making `Cleanup()` return the number of "states" that were removed. This way you are guaranteed to get the correct delta value.